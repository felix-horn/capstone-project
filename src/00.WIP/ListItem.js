import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import Checkbox from '@material-ui/core/Checkbox'
import { InputBase } from '@material-ui/core'

ListItem.propTypes = {
  value: PropTypes.string.isRequired,
  handleCheckboxClick: PropTypes.func.isRequired,
  isChecked: PropTypes.bool,
}

function ListItem({ value, handleCheckboxClick, isChecked }) {
  return (
    <StyledListItem>
      <Checkbox
        color="default"
        checked={isChecked}
        onChange={handleCheckboxClick}
      />

      <InputBase name={value} value={value} />
    </StyledListItem>
  )
}

const StyledListItem = styled.div`
  display: flex;
  align-items: center;
`

export default ListItem
