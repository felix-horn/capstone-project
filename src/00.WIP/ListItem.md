```js padded
<ListItem isChecked=""/>
<ListItem isChecked="" value="Butter"/>
<ListItem isChecked="true" value="Milk"/>
```