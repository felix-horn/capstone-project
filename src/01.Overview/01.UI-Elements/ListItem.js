import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

ListItem.propTypes = {
  title: PropTypes.string.isRequired,
}

export default function ListItem({ title }) {
  return (
    <ListItemLayout>
      <Checkbox />
      <Title>{title}</Title>
    </ListItemLayout>
  )
}

const ListItemLayout = styled.label`
  display: flex;
  align-items: center;
  gap: 5px;
`

const Checkbox = styled.div`
  border: 1px solid var(--light-gray);
  border-radius: 2px;
  height: 15px;
  width: 15px;
`

const Title = styled.p`
  margin: 0;
  color: var(--dark-gray);
`
