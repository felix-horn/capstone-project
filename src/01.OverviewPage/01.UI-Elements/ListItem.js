import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

ListItem.propTypes = {
  title: PropTypes.string.isRequired,
}

export default function ListItem({ title }) {
  return (
    <ListItemStyled>
      <CheckboxStyled />
      <TitleStyled>{title}</TitleStyled>
    </ListItemStyled>
  )
}

const ListItemStyled = styled.label`
  display: flex;
  align-items: center;
  gap: 5px;
`

const CheckboxStyled = styled.div`
  border: 1px solid var(--light-gray);
  border-radius: 2px;
  height: 15px;
  width: 15px;
`

const TitleStyled = styled.p`
  margin: 0;
  color: var(--dark-gray);
`
