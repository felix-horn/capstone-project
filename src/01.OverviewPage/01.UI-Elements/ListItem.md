```js padded
<ListItem title="Pommes frites"/>
<ListItem title="Ketchup"/>
```