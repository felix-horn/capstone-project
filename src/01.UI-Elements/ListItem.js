import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

ListItem.propTypes = {
  titleListItem: PropTypes.string.isRequired,
  isChecked: PropTypes.bool,
}

export default function ListItem({ titleListItem, isChecked }) {
  return (
    <ListItemStyled>
      <CheckboxStyled type="checkbox" checked={isChecked} />
      <TitleStyled>{titleListItem}</TitleStyled>
    </ListItemStyled>
  )
}

const ListItemStyled = styled.label`
  display: flex;
  align-items: center;
`
const CheckboxStyled = styled.input`
  margin-right: 10px;
  transform: scale(1.5);
`
const TitleStyled = styled.span`
  margin-left: 10px;
`
