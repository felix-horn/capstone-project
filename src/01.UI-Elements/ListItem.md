```js padded
<ListItem titleListItem="Milk"/>
<ListItem titleListItem="Butter" isChecked="true" />
```