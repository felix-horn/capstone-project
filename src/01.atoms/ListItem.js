import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

ListItem.propTypes = {
  titleListItem: PropTypes.string.isRequired,
  isChecked: PropTypes.bool,
  onCheckboxClick: PropTypes.func.isRequired,
}
function ListItem({ titleListItem, isChecked, onCheckboxClick }) {
  return (
    <LabelStyled>
      <Checkbox
        type="checkbox"
        checked={isChecked}
        onChange={onCheckboxClick}
      />
      <span>{titleListItem}</span>
    </LabelStyled>
  )
}

const LabelStyled = styled.label`
  display: block;
`

const Checkbox = styled.input`
  font-size: 1em;
`

/** @component */
export default ListItem
