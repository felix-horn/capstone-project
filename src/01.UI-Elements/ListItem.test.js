import { getByRole, render } from '@testing-library/react'
import ListItem from './ListItem'
import user from '@testing-library/user-event'

describe('ListItem', () => {
  it('shows the correct title', () => {
    const { getByText } = render(<ListItem title="Butter" />)

    expect(getByText('Butter')).toBeInTheDocument()
  })

  it('toggles the checkbox by property change', () => {
    const onToggleMock = jest.fn()
    const { rerender, getByLabelText } = render(
      <ListItem isChecked={true} title="Milk" onToggle={onToggleMock} />
    )
    const checkbox = getByLabelText('Milk')
    expect(checkbox).toHaveProperty('checked', true)

    rerender(
      <ListItem isChecked={false} title="Milk" onToggle={onToggleMock} />
    )
    expect(checkbox).toHaveProperty('checked', false)
  })

  it('calls onToggle', () => {
    const onToggleMock = jest.fn()
    const { getByText } = render(
      <ListItem onToggle={onToggleMock} title="Bread" isChecked />
    )

    const listItem = getByText('Bread')
    user.click(listItem)

    expect(onToggleMock).toHaveBeenCalled()
  })

  it('toggles the checkbox', () => {
    const onToggleMock = jest.fn()
    const { getByLabelText } = render(
      <ListItem title="Milk" onToggle={onToggleMock} />
    )
    const checkbox = getByLabelText('Milk')
    expect(checkbox).toHaveProperty('checked', false)

    user.click(checkbox)

    expect(checkbox).toHaveProperty('checked', true)
  })
})
