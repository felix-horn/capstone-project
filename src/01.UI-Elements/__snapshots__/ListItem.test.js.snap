// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem shows the right item title 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  margin-right: 10px;
  -webkit-transform: scale(1.5);
  -ms-transform: scale(1.5);
  transform: scale(1.5);
}

.c2 {
  margin-left: 10px;
}

<label
  class="c0"
>
  <input
    class="c1"
    type="checkbox"
  />
  <span
    class="c2"
  >
    Milk
  </span>
</label>
`;
